src/maxmind.cpp

```diff
@@ -175,23 +175,23 @@ CharacterVector maxmind_bindings::postcode(MMDB_s *data, CharacterVector ip_addr
 }
 
 CharacterVector maxmind_bindings::connection(MMDB_s *data, CharacterVector ip_addresses){
-  return mmdb_getstring(data, ip_addresses, "connection_type", NULL);
+  return mmdb_getstring(data, ip_addresses, "traits", "connection_type", NULL);
 }
 
 CharacterVector maxmind_bindings::isp(MMDB_s *data, CharacterVector ip_addresses){
-  return mmdb_getstring(data, ip_addresses, "isp", NULL);
+  return mmdb_getstring(data, ip_addresses, "traits", "isp", NULL);
 }
 
 CharacterVector maxmind_bindings::organization(MMDB_s *data, CharacterVector ip_addresses){
-  return mmdb_getstring(data, ip_addresses, "organization", NULL);
+  return mmdb_getstring(data, ip_addresses, "traits", "organization", NULL);
 }
 
 IntegerVector maxmind_bindings::asn(MMDB_s *data, CharacterVector ip_addresses){
-  return mmdb_getint32(data, ip_addresses, "autonomous_system_number", NULL);
+  return mmdb_getint32(data, ip_addresses, "traits", "autonomous_system_number", NULL);
```

Comment:
I don't think these changes would work with the MaxMind GeoIP2 ISP, ASN, or Connection-Type databases. They do not use the nested `traits` map.

Reply:
The original bug was for a "MaxMind Enterprise" database - which was also using the "traits" map.

How would you like issues between different layouts handled? I don't have access to other sources...

I've solved my issue, so I'm okay if its only a partial fix as I'm unlikely to use different database vendors...But making it clean for others would be good...

Reply:
There are [test databases](https://github.com/maxmind/MaxMind-DB/tree/main/test-data) for the types MaxMind provides.

Yeah, I am not exactly sure what the best fix is given what this code looks like. I guess one option is to check both locations. Another would be to check the database type and do the right thing based on that.

The official MaxMind libraries generally return a structured objects with separate lookup methods for the different database types or as nested maps for the lower-level APIs.